### PR TITLE
Bind stats

### PR DESCRIPTION
**Pull request by gcmalloc**
adding support for configuring bind stat api

### CODE REVIEW

File: manifests/server/options.pp [hunk 1]
```diff
@@ -70,6 +84,14 @@
   }
   validate_array($allow_query)
 
+  if !is_numeric($statistic_channel_port) {
+    fail('The statistic_channel_port is not a number')
+  }
+
+  if !validate_ip($statistic_channel_ip) {
```
Comment (solarkennedy) [inline manifests/server/options.pp]: I think validate_ip auto fails, like validate_array or whatever, The extra if is extra

File: templates/named.conf.options.erb [hunk 2]
```diff
@@ -62,6 +62,12 @@ check-names slave <%= @check_names_slave %>;
 check-names response <%= @check_names_response %>;
 <% end -%>
 
+<% if @statistic_channel_ip and @statistic_channel_ip -%>
```
Comment (solarkennedy) [inline templates/named.conf.options.erb]: This needs a test.

Can you check out the spec/ folder and find similar tests?

I would like to see that the inet bla lport bla line is not in there by default, but is in there when you set the right ip/port. I wouldn't bother testing validation.

Comment (gcmalloc) [inline templates/named.conf.options.erb]: ok, wanna have test for the other validation too ? I could add them in another pull request.

Comment (solarkennedy) [inline templates/named.conf.options.erb]: Na, I don't think we need to test the validation. I'm mostly concerned about the template string values that end up on disk.

**Comment by gcmalloc** [comment]
Sorry for the delay, should be good now.

**Comment by solarkennedy** [comment]
Looks great! Thank you for the follow through!

**Pull request merged**